crates/valence/src/tests/boss_bar.rs

```diff
@@ -1,47 +1,132 @@
 use bevy_app::App;
 use bevy_ecs::entity::Entity;
-use valence_boss_bar::components::{BossBar, BossBarColor, BossBarDivision, BossBarFlags};
-use valence_core::text::Text;
-use valence_instance::Instance;
+use valence_boss_bar::{components::{BossBarBundle, BossBarViewers, BossBarColor, BossBarDivision, BossBarFlags, BossBarTitle, BossBarHealth, BossBarStyle}, packet::BossBarS2c};
+use valence_core::{text::Text, despawn::Despawned};
 
-use super::{MockClientHelper, scenario_single_client, create_mock_client};
+use super::{MockClientHelper, scenario_single_client};
 
-/*#[test]
+#[test]
 fn test_intialize_on_join() {
     let mut app = App::new();
-    let (_, instance_ent) = prepare(&mut app);
+    let (client_ent, mut client_helper, instance_ent) = prepare(&mut app);
 
-    let (mut client, mut client_helper) = create_mock_client();
-    let client_ent = app.world.spawn(client).id();
+    let mut boss_bar = app.world.get_mut::<BossBarViewers>(instance_ent).unwrap();
+    boss_bar.current_viewers.push(client_ent);
 
-}*/
+    app.update();
+
+    let frames = client_helper.collect_sent();
+    frames.assert_count::<BossBarS2c>(1);
+}
+
+#[test]
+fn test_despawn() {
+    let mut app = App::new();
+    let (client_ent, mut client_helper, instance_ent) = prepare(&mut app);
+
+    let mut boss_bar = app.world.get_mut::<BossBarViewers>(instance_ent).unwrap();
+    boss_bar.current_viewers.push(client_ent);
+
+    app.update();
+
+    app.world.entity_mut(instance_ent).insert(Despawned);
+
+    app.update();
+
+    let frames = client_helper.collect_sent();
+    frames.assert_count::<BossBarS2c>(2);
```

Comment:
I don't know if there's a way to view the actual content of the packets. I'm just checking the count now.